extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,386 @@
+   * @param tracerName the name of the tracer to use
+   */
+  public Tracing(OpenTelemetry openTelemetry, String tracerName) {
+    this(openTelemetry, openTelemetry.getTracer(tracerName));
+  }
+
+  /**
+   * Creates a new instance of {@link Tracing}.
+   *
+   * @param openTelemetry the {@link OpenTelemetry} instance
+   * @param tracer the {@link Tracer} to use
+   */
+  public Tracing(OpenTelemetry openTelemetry, Tracer tracer) {
+    this.openTelemetry = openTelemetry;
+    this.tracer = tracer;
+  }
+
+  /**
+   * Creates a new {@link SpanBuilder} with the given span name.
+   *
+   * @param spanName the name of the span
+   * @return the {@link SpanBuilder}
+   */
+  public SpanBuilder spanBuilder(String spanName) {
+    return tracer.spanBuilder(spanName);
+  }
+
+  /**
+   * Runs the given {@link Runnable} in a new span with the given name.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanName the name of the span
+   * @param runnable the {@link Runnable} to run
+   */
+  public void run(String spanName, Runnable runnable) {
+    run(tracer.spanBuilder(spanName), runnable);
+  }
+
+  /**
+   * Runs the given {@link Runnable} inside of the span created by the given {@link SpanBuilder}.
+   * The span will be ended at the end of the {@link Runnable}.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param runnable the {@link Runnable} to run
+   */
+  public void run(SpanBuilder spanBuilder, Runnable runnable) {
+    call(
+        spanBuilder,
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  /**
+   * Runs the given {@link Callable} inside a new span with the given name.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanName the name of the span
+   * @param callable the {@link Callable} to call
+   * @param <T> the type of the result
+   * @return the result of the {@link Callable}
+   */
+  public <T> T call(String spanName, Callable<T> callable) {
+    return call(tracer.spanBuilder(spanName), callable);
+  }
+
+  /**
+   * Runs the given {@link Callable} inside of the span created by the given {@link SpanBuilder}.
+   * The span will be ended at the end of the {@link Callable}.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param callable the {@link Callable} to call
+   * @param <T> the type of the result
+   * @return the result of the {@link Callable}
+   */
+  public <T> T call(SpanBuilder spanBuilder, Callable<T> callable) {
```

Comment:
I'm not sure about hiding checked exceptions in these methods. I'm interested in hearing from @jack-berg and @jkwatson since the goal is for this module to align with OpenTelemetry API practices.

```suggestion
  public <T> T call(SpanBuilder spanBuilder, Callable<T> callable) throws Exception {
```

Reply:
changed using ThrowingSupplier as discussed